SportsHub/SportsHub.Core/Models/League.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace SportsHub.Domain.Models
+{
+    public class League
+    {
+        public League()
+        {
+            Teams = new HashSet<Team>();
```

Comment:
We have been asked to remove `ICollection` initializations from Model's constructors before, maybe consider doing the same here for consistency?